data-generator/src/main/java/com/github/streamshub/kafka/data/generator/schema/ClickStream.java

```diff
@@ -0,0 +1,267 @@
+/**
+ * Autogenerated by Avro
+ *
+ * DO NOT EDIT DIRECTLY
+ */
+package com.github.streamshub.kafka.data.generator.schema;
+
+import org.apache.avro.specific.SpecificData;
+
+@SuppressWarnings("all")
+@org.apache.avro.specific.AvroGenerated
+public class ClickStream extends org.apache.avro.specific.SpecificRecordBase implements org.apache.avro.specific.SpecificRecord {
```

Comment:
there's probably some arrangement where it gets generated to directory under `target` instead of being checked in, but would defer that to a later PR.

Reply:
hmm, that's interesting. I wonder if we would end up wanting to keep it being generated in the source directory but not checked in, similar to the Builder classes used in Strimzi. But yeah we can experiment. I could update it now to generate into the target directory and just copy/paste the schema String. The main reason I added the schema generation was to make it easier for people to add other data sources. What do you think?

Reply:
I think it's best to merge first then come back to it.

I wonder it would be better if the contract for `Data` was `SpecificRecord generate()` then in each Data implementation you'd be working with the typesafe, generated class, instead of a Map.

Reply:
I've created an issue for this https://github.com/streamshub/flink-sql-examples/issues/2